rust-runtime/smithy-client/src/dvr/replay.rs

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+use crate::dvr::{Action, ConnectionId, Direction, Event};
+use bytes::Bytes;
+use http::{Request, Version};
+use http_body::Body;
+use smithy_http::body::SdkBody;
+use std::collections::{HashMap, VecDeque};
+use std::error::Error;
+use std::sync::atomic::{AtomicUsize, Ordering};
+use std::sync::{Arc, Mutex};
+use std::task::{Context, Poll};
+
+/// Replay traffic recorded by a [`RecordingConnection`](RecordingConnection)
+#[derive(Clone, Debug)]
+pub struct ReplayingConnection {
+    events: Arc<Mutex<HashMap<ConnectionId, VecDeque<Event>>>>,
```

Comment:
Do we need a map of connection id -> events, or can this just be a vec of a vec?

Reply:
discussed offline, you could do it with Vec<Vec<..>> but it then becomes more annoying to reassemble the requests afterword